### PR TITLE
feat: pwcompose up now injects pwinit binary in containers before sta…

### PR DESCRIPTION
**Pull request by Z-a-r-a-k-i**
…rting them

### CODE REVIEW

File: go/pkg/pwcompose/compose.go [hunk 1]
```diff
@@ -235,14 +238,44 @@ func Up(
...
+	}
+
+	// copy pathwar binary inside all containers
+	pwInfo, err := GetPathwarInfo(ctx, cli)
+	if err != nil {
+		return errcode.ErrComposeGetPathwarInfo.Wrap(err)
+	}
+
+	for _, service := range pwInfo.RunningInstances {
+		if challengeID == fmt.Sprintf("%s@%s", service.Labels[challengeNameLabel], service.Labels[challengeVersionLabel]) {
```
Comment (moul) [inline go/pkg/pwcompose/compose.go]: You can create an helper for this:

```go
fmt.Sprintf("%s@%s", service.Labels[challengeNameLabel], service.Labels[challengeVersionLabel]
```

in order to call this:

```go
service.ChallengeID()
```

Comment (Z-a-r-a-k-i) [reply go/pkg/pwcompose/compose.go]: Something like this in config.go ?

`type service struct {
	ContainerName                             string            `yaml:"container_name,omitempty"`
	Image                                     string            `yaml:",omitempty"`
	Networks, Ports, Expose, Volumes, Command []string          `yaml:",omitempty"`
	VolumesFrom                               []string          `yaml:"volumes_from,omitempty"`
	DependsOn                                 []string          `yaml:"depends_on,omitempty"`
	CapAdd                                    []string          `yaml:"cap_add,omitempty"`
	Build                                     string            `yaml:",omitempty"`
	Restart                                   string            `yaml:",omitempty"`
	Environment                               map[string]string `yaml:",omitempty"`
	Labels                                    map[string]string `yaml:"labels,omitempty"`
}

func (s service) ChallengeID() string {
	return fmt.Sprintf("%s@%s", s.Labels[challengeNameLabel], s.Labels[challengeVersionLabel])
}`

Comment (Z-a-r-a-k-i) [reply go/pkg/pwcompose/compose.go]: Done

File: go/pkg/pwcompose/compose.go [hunk 2]
```diff
@@ -195,7 +198,7 @@ func Up(
 		service.ContainerName = fmt.Sprintf("%s.%s.%s.%s", challengeName, serviceName, imageHash[:6], instanceKey)
 		service.Restart = "unless-stopped"
 		service.Labels[instanceKeyLabel] = instanceKey
-		challengeID = fmt.Sprintf("%s@%s", service.Labels[challengeNameLabel], service.Labels[challengeVersionLabel])
+		challengeID = challengeIDFormatted(service.Labels[challengeNameLabel], service.Labels[challengeVersionLabel])
```
Comment (moul) [inline go/pkg/pwcompose/compose.go]:
```suggestion
		challengeID = challengeIDFormatted(service)
```

**Pull request merged**